src/constants_manager.cpp

```diff
@@ -0,0 +1,87 @@
+#include "constants_manager.h"
+
+#include "debug.h"
+
+#include <map>
+#include <algorithm>
+
+
+constants_manager &get_constants_manager()
+{
+    static constants_manager single_instance;
+    return single_instance;
+}
+
+
+constants_manager::game_constant &constants_manager::get_constant( const std::string &id )
```

Comment:
This function should be `const`. And returning a reference to a non-const `game_constant` looks strange. Does it need to be non-const?